### PR TITLE
Fix facility code resolution 

### PR DESCRIPTION
**Pull request by grahammcvea-defra**
Refactor facility code resolution, add default f null and add full test suite
- If facility code is null or empty then default SiteType to "AH"
- Introduce functional Result<T> type for error handling and composition.
- Refactor SiteTypeDerivedCodeLookupService to use Result<T> and split logic into smaller, testable methods.
- Fix substring matching bug by filtering out partial code matches, ensuring only the most specific codes are used.
- Add detailed logging for resolution failures with error codes.
- Add comprehensive unit tests covering edge cases, including substring overlaps, multiple codes, case insensitivity, and deduplication.
- Improve maintainability and robustness of facility code resolution logic.

### CODE REVIEW

File: src/KeeperData.Application/Services/SiteTypeDerivedCodeLookupService.cs [hunk 1]
```diff
@@ -1,78 +1,165 @@
...
+        if (hits.Count <= 1)
+            return Result.Ok(hits);
+
+        var filteredHits = new List<FacilityBusinessActivityMapDocument>();
+
+        foreach (var hit in hits)
+        {
+            var isPartialMatch = hits.Any(otherHit =>
+                otherHit != hit &&
+                otherHit.FacilityActivityCode!.Contains(hit.FacilityActivityCode!, StringComparison.OrdinalIgnoreCase));
```
Comment (HussRasit) [inline src/KeeperData.Application/Services/SiteTypeDerivedCodeLookupService.cs:79]: I don't thinks this contains will return NO_MAPPING_FOUND as we have two lots of codes that start with "LAF-HAT-NA" checking for length first with 

otherHit.FacilityActivityCode!.Length > hit.FacilityActivityCode!.Length && 

should fix this

**Review by HussRasit — CHANGES_REQUESTED** [review]

**Review by HussRasit — APPROVED** [review]
